### DISCUSSION TITLE
Report Generation

### DISCUSSION THREAD

**Discussion by silicahd** [nocodb/nocodb#710] [Features]
A really interesting feature in AppSheet is the ability to create a bot and reports. This would be an amazing feature.

**Comment by o1lab** [comment]
Can you help describe the usecase of a bot and reports.

**Reply by silicahd** [reply]
To be honest the best way is to create an appsheet and see how it works. 

In simple terms you can create a bot that accepts an action. 

Example: You can create a ticket system when the filed is set to complete, the bot will generate a report based on a template with the fields you would like. 

In my use case I would like to use this for field service to fix medical devices. Each incident is a ticket and requires a form to be completed with tasks. Than the form has to be in a pdf format with a signature and set to the customer. And to be more clear each device type has different tasks to make it more complicated.  But that is my use case. 

So in appsheet: I can create templates for each model. Than when i close a ticket if my form has typeA and it is complete it will generate the report based on typeA template with the data. 

This can be used for literality any application that requires reports to be sent to customers.

**Reply by julianstirling** [reply]
For report, I am pretty sure @silicahd is suggesting creating a document (probably a PDF) which can be used within a business.

There is the example of the medical device service record given above. Another one is if someone creates their accounting system in NoCodb there are many reports they might want to be able to generate.

* Each quarter or each month they might want to generate a document that lists some summaries of turnover, salary costs, etc, etc
* Thinking from a charity/non-profit perspective, a grant funder may ask for a list of all items/services purchased with their money.
* Creating invoices for customers

I can see creating full document generation may be scope creep for nocodb, but perhaps it may be less so if a library like [pdfme](https://github.com/pdfme/pdfme) was used. They even have a UI for creating PDF templates with set fields:

https://pdfme.com/demo
